### DISCUSSION TITLE
Running on selected time segments of an audio

### DISCUSSION THREAD

**Discussion by patarapolw** [openai/whisper#135] [Q&A] [answered]
I wonder there is a good and easy way to do?

Running with `small` model sometimes generates text segments with repeated text for a relatively silent audio segment.

![image](https://user-images.githubusercontent.com/21255931/192185699-5e075556-f6a4-4564-a487-1f8a97c4c5f8.png)

Otherwise, some audio lengths refused to be transcribed, but are generated repeated unrelated texts instead.

![image](https://user-images.githubusercontent.com/21255931/192206832-90d46af6-4591-49a9-a793-142102ff7df4.png)

So, I am thinking of doing a second run, perhaps with larger models. Maybe I can cut into audio segments, but they would be too short, and no surrounding audio's optimatization. Also, 30 seconds won't be met.

**Comment by jongwook** [comment, marked as answer]
This package doesn't support for this as-is, but you could manually create segmented files using `ffmpeg`, e.g.

```bash
ffmpeg -i input.mp3 -ss 2:20.00 -to 3:40.00 output.mp3
```

**Reply by patarapolw** [reply]
BTW, can whisper be piped into? Like `ffmpeg -i input.mp3 -ss 2:20.00 -to 3:40.00 >> whisper`? (I am on Windows, but I am not used to Powershell or cmd.)

**Reply by jongwook** [reply]
I didn't intend this to work, but turns out this works in bash:

```bash
ffmpeg -i test.mp3 -ss 2:20.00 -to 3:40.00 -f wav - | whisper -
```

**Reply by patarapolw** [reply]
Thanks. Exact command works in Powershell as well.

EDIT: No, it doesn't.

![image](https://user-images.githubusercontent.com/21255931/192214099-6c3a041a-dc64-4c7a-953f-3ac06d8212fe.png)

**Reply by jongwook** [reply]
I think the segmentation worked and the model is just struggling to transcribe the audio. You may find better results with `medium` or `large` model if you're using the default `medium` model, especially for non-English.

**Reply by patarapolw** [reply]
> `medium` or `large` mode

How do I allocate memory? Both medium and large crashed.

> RuntimeError: CUDA out of memory. Tried to allocate 20.00 MiB (GPU 0; 4.00 GiB total capacity; 3.48 GiB already allocated; 0 bytes free; 3.49 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

![image](https://user-images.githubusercontent.com/21255931/192219173-7274fa50-2638-45a1-855f-3d06f010d638.png)

**Reply by jongwook** [reply]
Unfortunately, you'll likely need a different video card with larger dedicated GPU memory.. You can probably run inference on CPU (by giving `--device cpu`), but it'll be terribly slow.

**Reply by patarapolw** [reply]
nvm, I have another machine that can run `medium`, but I don't think `large` is an option. (Though, painfully slow 1:20.00 minutes.)

![image](https://user-images.githubusercontent.com/21255931/192234389-28400ecd-c752-48f8-93b6-7aca62b8f902.png)

Though, `small` runs faster on (newer) laptop than (older) PC.

**Comment by octimot** [comment]
FWIW I have created this example method to select and exclude time intervals from the audio file: https://github.com/openai/whisper/discussions/467
